cea/datamanagement/database_migrator.py

```diff
@@ -0,0 +1,147 @@
+"""
+This script checks a scenario for v2.29.0 format and migrates the input tables it to the v2.31.1 format.
+
+NOTE: You'll still need to run the archetypes-mapper after this script has run.
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+import os
+import cea
+import pandas as pd
+import collections
+
+import cea.config
+import cea.inputlocator
+from cea.utilities.dbf import dbf_to_dataframe, dataframe_to_dbf
+
+__author__ = "Daren Thomas"
+__copyright__ = "Copyright 2020, Architecture and Building Systems - ETH Zurich"
+__credits__ = ["Daren Thomas"]
+__license__ = "MIT"
+__version__ = "0.1"
+__maintainer__ = "Daren Thomas"
+__email__ = "cea@arch.ethz.ch"
+__status__ = "Production"
+
+
+def find_migrators(scenario):
+    """
+    Add new migrations here as they become necessary
+    the database-migrator will run these in sequence starting from the first migrator found
+    (NOTE: I've added a dummy migration - 2.31 - 2.31.1 - to show how the principle works)
+    """
+    migrations = collections.OrderedDict()
+    migrations["v2.29.0 - v2.31.0"] = (is_2_29, migrate_2_29_to_2_31)
+    migrations["v2.31.0 - v2.31.1"] = (is_2_31, migrate_2_31_to_2_31_1)
+
+    for key, migration_info in migrations.items():
+        identifier, migrator = migration_info
+        if identifier(scenario):
+            yield key, migrator
+
+
+def is_2_29(scenario):
+    if not os.path.exists(os.path.join(scenario, "inputs", "building-properties", "age.dbf")):
+        return False
+    if not os.path.exists(os.path.join(scenario, "inputs", "building-properties", "occupancy.dbf")):
+        return False
+    if os.path.exists(os.path.join(scenario, "inputs", "building-properties", "typology.dbf")):
+        # avoid migrating multiple times
+        return False
+    return True
+
+
+def migrate_2_29_to_2_31(scenario):
+    def lookup_standard(year, standards_df):
+        # find first standard that is similar to the year
+        standard = standards_df[(standards_df.YEAR_START < year) & (year < standards_df.YEAR_END)].iloc[0]
```

Comment:
I'm getting `IndexError: single positional indexer is out-of-bounds` when `year` is 1971. I guess it should be `<=` instead of  `<` and handle the exception when the dataframe is empty